### PR TITLE
throttle live based on connections

### PR DESCRIPTION
**Pull request by christianalfoni**

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/3664/commits/7caf8a90a2fc86612ba20d1c4fdc15d875c44431 is at https://pr3664.build.csb.dev/s/new.

File: packages/app/src/app/overmind/namespaces/live/actions.ts [hunk 1]
```diff
@@ -21,6 +21,23 @@ export const signInToRoom: AsyncAction<{
...
+  code: string;
+  revision: number;
+  saved_code: string;
+}> = ({ state, effects }, { moduleShortid, revision, code, saved_code }) => {
+  effects.live.resetClient(moduleShortid, revision);
+  const module = state.editor.currentSandbox.modules.find(
+    moduleItem => moduleItem.shortid === moduleShortid
+  );
+
+  module.code = code;
```
Comment (CompuIves) [inline packages/app/src/app/overmind/namespaces/live/actions.ts]:
```suggestion
if (code !== undefined) {
  module.code = code;
}
```

File: packages/app/src/app/overmind/namespaces/live/actions.ts [hunk 2]
```diff
@@ -21,6 +21,23 @@ export const signInToRoom: AsyncAction<{
...
+  module.savedCode = saved_code;
```
Comment (CompuIves) [inline packages/app/src/app/overmind/namespaces/live/actions.ts]:
```suggestion
if (saved_code !== undefined) {
  module.savedCode = saved_code;
}
```

File: packages/app/src/app/overmind/effects/live/clients.ts [hunk 3]
```diff
@@ -54,6 +51,12 @@ class CodeSandboxOTClient extends OTClient {
   serverAck() {
     try {
       super.serverAck();
+
+      if (this.state === synchronized_) {
+        const pending = this.pending;
+        this.pending = null;
+        pending.resolve();
```
Comment (CompuIves) [inline packages/app/src/app/overmind/effects/live/clients.ts]: Is there a chance that this is null?

**Pull request merged**